dotcom-rendering/src/components/ExpandableMarketingCardSwipeable.tsx

```diff
@@ -0,0 +1,245 @@
+
+	// The distance to the left from the pointer down and current pointer, whilst the pointer is still down.
+	const [dragLeftDistance, setDragLeftDistance] = useState(0);
+
+	useEffect(() => {
+		setTopOfBody(document.querySelector('[data-gu-name="body"]'));
+	}, []);
+
+	useEffect(() => {
+		if (!topOfBody) return;
+
+		/**
+		 * It's possible that the viewport does not start at the top of the page.
+		 */
+		if (isViewportBelowTopOfBody(topOfBody)) {
+			setShouldDisplayCard(true);
+			return;
+		}
+
+		/**
+		 * Show the card when the top of the body moves out of the viewport.
+		 */
+		const observer = new window.IntersectionObserver(
+			([entry]) => {
+				if (!entry) return;
+				if (entry.isIntersecting) {
+					setShouldDisplayCard(true);
+				}
+			},
+			{ rootMargin: '0px 0px -100%' },
+		);
+
+		observer.observe(topOfBody);
+
+		return () => {
+			observer.disconnect();
+		};
+	}, [topOfBody]);
+
+	if (!shouldDisplayCard) {
+		return null;
+	}
+
+	if (!isExpanded) {
+		return (
+			<div css={[stickyContainerStyles]}>
+				<div
+					css={[
+						absoluteContainerStyles,
+						isDown &&
+							css`
+								left: ${dragLeftDistance}px;
+							`,
+						hasSwipedLeft && hideBannerStyles,
+					]}
+					role="button"
+					tabIndex={0}
+					onKeyDown={(event) => {
+						if (event.key === 'Enter') {
+							setIsExpanded(true);
+						}
+						if (event.key === 'Escape') {
+							setIsClosed(true);
+						}
+					}}
+					onPointerDown={(event) => {
+						event.preventDefault();
+
+						setIsDown(true);
+
+						setLastDownXCoord(event.clientX);
+						setLastDownYCoord(event.clientY);
+					}}
+					onPointerUp={(event) => {
+						event.preventDefault();
+
+						setIsDown(false);
+
+						if (
+							isClick(
+								[event.clientX, event.clientY],
+								[lastDownXCoord, lastDownYCoord],
+							)
+						) {
+							setIsExpanded(true);
+						} else if (isLeftSwipe(event.clientX, lastDownXCoord)) {
+							setHasSwipedLeft(true);
+						}
+					}}
+					onPointerMove={(event) => {
+						event.preventDefault();
+
+						if (isDown) {
+							setDragLeftDistance(
+								Math.min(
+									0,
+									event.clientX - (lastDownXCoord ?? 0),
+								),
+							);
+						}
+					}}
+					onTouchStart={(event) => {
+						event.preventDefault();
+					}}
+				>
+					<ExpandableMarketingCard
+						{...{
+							guardianBaseURL,
+							heading,
+							kicker,
+							isExpanded,
+							setIsExpanded,
+							setIsClosed,
+						}}
+					/>
+				</div>
+			</div>
+		);
+	}
+
+	return (
+		<div css={[stickyContainerStyles]}>
+			<div
+				css={[absoluteContainerStyles]}
+				role="button"
+				tabIndex={0}
+				onKeyDown={(event) => {
+					if (event.key === 'Escape') {
+						setIsClosed(true);
+					}
+				}}
+			>
+				<ExpandableMarketingCard
+					{...{
+						guardianBaseURL,
+						heading,
+						kicker,
+						isExpanded,
+						setIsExpanded,
+						setIsClosed,
+					}}
+				/>
```

Comment:
Is the wrapped object and internal spread intentional?

I think it is equivalent without the internal spread:

However regular props are potentially less surprising and more idiomatic for future travellers:


```suggestion
				<ExpandableMarketingCard
						guardianBaseURL={guardianBaseURL}
						heading={heading}
						kicker={kicker}
						isExpanded={isExpanded}
						setIsExpanded={setIsExpanded}
						setIsClosed={setIsClosed}
				/>
```

Reply:
That's a really good point. Will update